lib/tasks/truncate_data.rake

```diff
@@ -0,0 +1,24 @@
+namespace :ofn do
+  namespace :data do
+    desc 'Truncate data'
+    task truncate: :environment do
+      guard_and_warn
+
+      # truncate Orders older than 6 months
+      Spree::Order
+        .where("completed_at < (?)", 6.months.ago)
+        .destroy_all
+
+      # truncate OrderCycles older than 6 months
+      OrderCycle
+        .where("created_at < ?", 6.months.ago)
+        .destroy_all
+
+      # truncate Products not in an OC and older than 6 months
+      Spree::Product
+        .with_order_cycles_outer
+        .where('o_order_cycles.id IS NULL and spree_products.created_at < (?)', 6.months.ago)
+        .destroy_all
+    end
```

Comment:
I don't understand why we need this truncation with a 6-month window. Can you provide more details in a code comment maybe?

Reply:
I have added a comment at the top of the task